### DISCUSSION TITLE
Adding User to CMS User registered through manual Form

### DISCUSSION THREAD

**Discussion by aashish803** [OrchardCMS/OrchardCore#15561] [Q&A]
Hello everyone, 
          
![image](https://github.com/OrchardCMS/OrchardCore/assets/144659637/3e0e531c-9ec1-4478-91a9-d017864646c0)

Above is the manual registration form created from the orchard core which is submitting details in different DB. I want to add these details of user to Orchard DB as a new user.
I.E  when we create any user from the F.E form the details of that particular user should be displayed in orchard CMS users
![image](https://github.com/OrchardCMS/OrchardCore/assets/144659637/e36ffd6c-2e00-4a1b-b4f8-ebcd7b1c874c). 
Please let me know if you want more details. 

Thanks,
Regards

**Comment by Piedone** [comment]
You'll need to extend the `User` entity, in a way a bit similar to how you can extend a content type with a custom content part. The Demo module has an example of this: you can explore it by starting from [`UserProfile`](https://github.com/OrchardCMS/OrchardCore/blob/42d6214aa8de9b3186d1348661d8f9bc2b4e4020/src/OrchardCore.Modules/OrchardCore.Demo/Models/UserProfile.cs). I suggest cloning the solution and looking up its references.

**Reply by aashish803** [reply]
Thanks for your response, but the method you've provided is adding fields in the CMS ( below is the reference )
![image](https://github.com/OrchardCMS/OrchardCore/assets/144659637/3e9dd34a-1d56-48dd-bc7d-2cd6dbf4f649)

I want the response of the registeration form as the list in  the CMS users.

**Reply by Piedone** [reply]
Sorry, I don't understand.

**Comment by hishamco** [comment]
You can extend the User entity as @Piedone said before, or use the `Properties` which can hold the extra properties as JSON

**Reply by aashish803** [reply]
Hey @hishamco and @Piedone 
I don't think you've understood me correctly. Let me clarify it once more.

I have developed a user registration form using a CMS on the frontend (FE). Now, when I fill in all the details in the form and submit it, all the data should be sent to the CMS user area. This way, I can also utilize those frontend users within the CMS. Currently, in the CMS's user admin menu, we only see the list of users created within the CMS itself. However, we aim to integrate the frontend registered users into the CMS as well.

Thanks,

**Reply by hishamco** [reply]
I don't there's no notion of FE users but you could create a role then all the FE users can be added into

What I mentioned before is a valid use case, I already implemented such thing in one of the websites in the past

**Reply by aashish803** [reply]
![Screenshot (95)](https://github.com/OrchardCMS/OrchardCore/assets/144659637/2f87ac6d-5f7b-4281-acf0-2b081a4ad410)
Above is the manual register form made using Content Types

Now I want that after submitting this form user detail get added to or new user get registered and show here.
![image](https://github.com/OrchardCMS/OrchardCore/assets/144659637/d195d3b2-8ed2-4c94-8725-b31a7fb9feda)

If it is possible can you share the use case of the URL which you have implemented in your websites. 

Let me know if we are on the same page or not?
Thanks.

**Reply by hishamco** [reply]
As I said before there different ways:

1. Extend user entity
2. Use Properties to save a custom user properties as JSON
3. Use Custom User Settings

Of course you may need to override the UI to display the additional properties

**Reply by Piedone** [reply]
Is your question then only about how to display the new properties in the users list? You'll need to display a shape from your driver's `Display()`, targeting the `SummaryAdmin` display type. Search for "SummaryAdmin" in the codebase for examples, or https://github.com/Lombiq/Orchard-Training-Demo-Module for a complete, from the ground up tutorial on how to work with shapes.

**Reply by aashish803** [reply]
Hey

I have tried implementing using Workflows and I achieved the desired output, thanks your guidance.

**Reply by Piedone** [reply]
Great, then! Please mark an answer.

**Reply by hishamco** [reply]
@aashish803 please elaborate
